src/main/java/it/gov/pagopa/cgn/portal/service/DiscountService.java

```diff
@@ -221,6 +219,47 @@ private ProfileEntity validateDiscount(String agreementId, DiscountEntity discou
         ProfileEntity profileEntity = profileService.getProfile(agreementId)
                 .orElseThrow(() -> new InvalidRequestException("Cannot create discount without a profile"));
 
+        commonDiscountValidation(profileEntity, discountEntity, isBucketFileChanged);
+
+        ValidationUtils.performConstraintValidation(factory.getValidator(), discountEntity);
+        return profileEntity;
+    }
+
+    private void validatePublishingDiscount(AgreementEntity agreementEntity, DiscountEntity discount) {
+        ProfileEntity profileEntity = profileService.getProfile(agreementEntity.getId())
+                .orElseThrow(() -> new InvalidRequestException("Cannot get discount's profile"));
+
+        // perform common discount validation to keep entities coherent
+        commonDiscountValidation(profileEntity, discount, false);
+
+        //perform publishing specific validation
+        if (profileEntity.getDiscountCodeType().equals(DiscountCodeTypeEnum.BUCKET)
+                && (discount.getLastBucketCodeLoad() == null || bucketService.isLastBucketLoadStillLoading(discount.getLastBucketCodeLoad().getId()))) {
+            throw new ConflictErrorException("Cannot publish a discount with a bucket load in progress");
+        }
+        if (!AgreementStateEnum.APPROVED.equals(agreementEntity.getState())) {
+            throw new InvalidRequestException("Cannot publish a discount with a not approved agreement");
+        }
+        if (DiscountStateEnum.SUSPENDED.equals(discount.getState())) {
```

Comment:
not sure this is correct. When a discount is suspended we have to set status to draft for replying publish operation?

Reply:
Isn't possible to re-publish a suspended discount after having modified it?

Reply:
yes it is, but when users asks to modify a suspended discount it doesn't trigger a state change?